### PR TITLE
[SYCL] Add support for ext_intel_matrix aspect

### PR DESCRIPTION
**Pull request by lbushi25**
This PR adds the aspect for the sycl_ext_intel_matrix extension and allows code to verify whether the device supports the advanced matrix API. The aspect has been added to the aspect definition file and the corresponding macro has been added to the feature test file set to 1 meaning supported.

### CODE REVIEW

**Review by maksimsab — APPROVED** [review]
Tools part LGTM.

nit: I would suggest `ext_intel_matrix` name of the aspect in the PR title instead of `sycl_ext_intel_matrix`.

**Review by againull — COMMENTED** [review]
Please add a test.

File: sycl/source/detail/device_impl.cpp [hunk 1]
```diff
@@ -567,6 +567,22 @@ bool device_impl::has(aspect Aspect) const {
...
   }
+  case aspect::ext_intel_matrix: {
+    using arch = sycl::ext::oneapi::experimental::architecture;
+    const std::vector<arch> supported_archs = {
+        arch::intel_cpu_spr, arch::intel_gpu_pvc, arch::intel_gpu_dg2_g10,
+        arch::intel_gpu_dg2_g11, arch::intel_gpu_dg2_g12};
+    try {
+      return std::any_of(
+          supported_archs.begin(), supported_archs.end(), [=](const arch a) {
+            return const_cast<device_impl *>(this)->extOneapiArchitectureIs(a);
```
[the remark this answers is not in the document]
Comment (lbushi25) [reply sycl/source/detail/device_impl.cpp:578]: Ok.

**Review by againull — APPROVED** [review]

**Comment by lbushi25** [comment]
@intel/llvm-gatekeepers This is good to go for merging!

**Pull request merged**